frame/support/test/tests/issue2219.rs

```diff
@@ -83,102 +79,108 @@ mod module {
 		}
 	}
 
-	pub trait Config: system::Config + TypeInfo {}
-
-	frame_support::decl_module! {
-		pub struct Module<T: Config> for enum Call where origin: T::RuntimeOrigin, system=system {}
-	}
-
-	#[derive(Encode, Decode, Copy, Clone, Serialize, Deserialize)]
-	pub struct Data<T: Config> {
-		pub data: T::BlockNumber,
-	}
-
-	impl<T: Config> Default for Data<T> {
-		fn default() -> Self {
-			Self { data: T::BlockNumber::default() }
-		}
+	#[pallet::pallet]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + TypeInfo {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {}
+
+	/// requirements to enter and maintain status in roles
+	#[pallet::storage]
+	#[pallet::getter(fn parameters)]
+	pub type Parameters<T: Config> =
+		StorageMap<_, Blake2_128Concat, Role, RoleParameters<T>, OptionQuery>;
+
+	/// the roles members can enter into
+	#[pallet::storage]
+	#[pallet::getter(fn available_roles)]
+	#[pallet::unbounded]
+	pub type AvailableRoles<T: Config> = StorageValue<_, Vec<Role>, ValueQuery>;
+
+	/// Actors list
+	#[pallet::storage]
+	#[pallet::getter(fn actor_account_ids)]
+	#[pallet::unbounded]
+	pub type ActorAccountIds<T: Config> = StorageValue<_, Vec<T::AccountId>>;
+
+	/// actor accounts associated with a role
+	#[pallet::storage]
+	#[pallet::getter(fn account_ids_by_role)]
+	#[pallet::unbounded]
+	pub type AccountIdsByRole<T: Config> = StorageMap<_, Blake2_128Concat, Role, Vec<T::AccountId>>;
+
+	/// tokens locked until given block number
+	#[pallet::storage]
+	#[pallet::getter(fn bondage)]
+	pub type Bondage<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, T::BlockNumber>;
+
+	/// First step before enter a role is registering intent with a new account/key.
+	/// This is done by sending a role_entry_request() from the new account.
+	/// The member must then send a stake() transaction to approve the request and enter the desired
+	/// role. The account making the request will be bonded and must have
+	/// sufficient balance to cover the minimum stake for the role.
+	/// Bonding only occurs after successful entry into a role.
+	#[pallet::storage]
+	#[pallet::getter(fn role_entry_requests)]
```

Comment:
I will keep the getters in the tests as we should remove them here last IMHO as to not accidentally break them.